### PR TITLE
Odilkhan Yakubov commits

### PR DESCRIPTION
**Pull request by odil24**
[Fix: Export Hair with Children particles](https://github.com/LuxCoreRender/BlendLuxCore/commit/e558dcb13374c40fc97d1679e68939eb43b610c2)

[Improved: Mesh export speed increased and cleanup the code](https://github.com/LuxCoreRender/BlendLuxCore/commit/6e5659beecc7e7478ca3f35813f4733fc7aa3edb)

[Improved: Increased export speed of Interactive rendering through camera](https://github.com/LuxCoreRender/BlendLuxCore/commit/dae9b9404163254fac43aab4d26f7b0bc77cec12)

[Fix: Cycles Transmission socket issue](https://github.com/LuxCoreRender/BlendLuxCore/commit/f82adc1d90fcc157a5e61256f12d3fd27d4edabb)

### CODE REVIEW

**Comment by odil24** [comment]
@neo2068 Please, review changes

**Review by neo2068 — APPROVED** [review]
This fixes issue #880

**Comment by odil24** [comment]
> This fixes issue #880 

Thanks. Is every commits are merged so?

**Comment by odil24** [comment]
Merged with fixed conflicts

**Comment by odil24** [comment]
This commits did not fixed an issue with Cycles node reader. Can you check @neo2068 ?
https://github.com/LuxCoreRender/BlendLuxCore/commit/f82adc1d90fcc157a5e61256f12d3fd27d4edabb

**Comment by neo2068** [comment]
> This commits did not fixed an issue with Cycles node reader. Can you check @neo2068 ? [f82adc1](https://github.com/LuxCoreRender/BlendLuxCore/commit/f82adc1d90fcc157a5e61256f12d3fd27d4edabb)

The patch for the Cycles node reader isn't correct. LuxCore disney material has only float parameter like in the original disney shader. Blender adapted the shader to have color parameters for specular tint and sheen tint. This cannot be transferred directly to LuxCore disney shader.

**Comment by odil24** [comment]
> Исправление для считывателя узлов Cycles неверно. Материал LuxCore disney имеет только параметр float, как и в оригинальном шейдере disney. Blender адаптировал шейдер для цветовых параметров для зеркального оттенка и оттенка блеска. Это не может быть передано напрямую в шейдер LuxCore disney.

are you have an idea how to fix that?

**Comment by odil24** [comment]
@neo2068 

https://github.com/LuxCoreRender/BlendLuxCore/assets/64298012/650741b4-23da-4494-9081-5a6968f1a870

**Review by neo2068 — APPROVED** [review]

**Comment by odil24** [comment]
@neo2068 review

**Review by neo2068 — APPROVED** [review]

File: export/mesh_converter.py [hunk 1]
```diff
@@ -22,22 +23,40 @@ def get_custom_normals_fast(mesh):
...
+def get_custom_normals_parallel(mesh):
+    if not mesh.has_custom_normals:
+        return None
+
+    loop_tri_count = len(mesh.loop_triangles)
+    custom_normals = np.zeros((loop_tri_count, NORMALS_ARRAY_SIZE), dtype=np.float32)
+
+    def process_loop(i, loop_tri):
+        split_normals = loop_tri.split_normals
+        custom_normals[i, :] = split_normals.flatten()
```
Comment (neo2068) [inline export/mesh_converter.py:39]: split_normals is a bpy.prop.array type and has no method flatten()

**Review by neo2068 — COMMENTED** [review]
Nice approach but when I change this code to a parallel version of the initial slow version, the run time for my test scene is 10 times slower.

**Comment by odil24** [comment]
@neo2068 so are you fixed now?

**Comment by neo2068** [comment]
> @neo2068 so are you fixed now?

@odil24 I reactivated the old slow export for custom normals for now. Can you check if it works for you?

**Comment by odil24** [comment]
> > @neo2068 so are you fixed now?
> 
> @odil24 I reactivated the old slow export for custom normals for now. Can you check if it works for you? 

Ok. I will take a look to this

**Comment by odil24** [comment]
@neo2068 review now

**Pull request closed without merging**